### PR TITLE
lockbook: init at 0.9.15

### PR DESCRIPTION
**Pull request by Parth**
👋 [Lockbook](https://lockbook.net) is an [open source](https://github.com/lockbook/lockbook) secure note taking app. This packages our CLI. Soon we'll have another pr for our desktop app. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by smailbarkouch — APPROVED** [review]

**Review by pluiedev — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs ❤️ Some changes that have to be done:

**Review by donovanglover — COMMENTED** [review]
Thank you for contributing to nixpkgs!

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16771,6 +16771,11 @@
     githubId = 33826198;
     name = "Philipp Arras";
   };
+  parth = {
+    github = "parth";
+    githubId = 821972;
+    name = "Parth Mehrotra";
+  };
```
[the remark this answers is not in the document]
Comment (Parth) [reply maintainers/maintainer-list.nix:16778]: separated out: [6d0eadc](https://github.com/NixOS/nixpkgs/pull/358794/commits/6d0eadc43951dcc0730fc18b34a6494f30251d58)

File: pkgs/by-name/lo/lockbook/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage {
+  pname = "lockbook";
+  version = "0.9.15";
+
+  src = fetchFromGitHub {
+    owner = "lockbook";
+    repo = "lockbook";
+    rev = "0.9.15";
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 4e5a5527502084439ec8d07167a5c1c987110533

File: pkgs/by-name/lo/lockbook/package.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage {
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

File: pkgs/by-name/lo/lockbook/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  doCheck = false;
+  cargoBuildFlags = [
+    "--package"
+    "lockbook-cli"
+  ];
+
+  # Optional: Declare metadata
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

File: pkgs/by-name/lo/lockbook/package.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    description = "CLI for Lockbook: Encrypted Notebook.";
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

File: pkgs/by-name/lo/lockbook/package.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  postInstall = ''
+    # Generate and install shell completions
+    mkdir -p $out/share/bash-completion/completions
+    mkdir -p $out/share/zsh/site-functions
+    mkdir -p $out/share/fish/vendor_completions.d
+
+    # Generate the completion scripts
+    $out/bin/lockbook completions bash > $out/share/bash-completion/completions/lockbook
+    $out/bin/lockbook completions zsh > $out/share/zsh/site-functions/_lockbook
+    $out/bin/lockbook completions fish > $out/share/fish/vendor_completions.d/lockbook.fish
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

File: pkgs/by-name/lo/lockbook/package.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+
+    outputHashes = {
+      "lb-fonts-0.1.5" = "sha256-WPUj+vn/E7uZrYvrpDonHQawEggKu+nvE0CVBPbDtyM=";
+      "lb-pdf-0.2.3" = "sha256-1Fj1qGxtZnUIwH+x2XZTXYufc+ECcr5/P+6ERuc8PvU=";
+      "minidom-0.15.3" = "sha256-V3Xy7r3eBheMlvVpGC/M/lTS7sM0C3L7ATRoeuM5c2A=";
+    };
+  };
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

File: pkgs/by-name/lo/lockbook/package.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  doCheck = false;
+  cargoBuildFlags = [
+    "--package"
+    "lockbook-cli"
+  ];
+
+  # Optional: Declare metadata
+  meta = with lib; {
+    description = "CLI for Lockbook: Encrypted Notebook.";
+    longDescription = "Write notes, sketch ideas, and store files in one secure place. Share seamlessly, keep data synced, and access it on any platform—even offline. Lockbook encrypts files so even we can’t see them, but don’t take our word for it: Lockbook is 100% open-source.";
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

File: pkgs/by-name/lo/lockbook/package.nix [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    homepage = "https://lockbook.net";
+    license = licenses.unlicense;
+    platforms = platforms.all;
+    changelog = "https://github.com/lockbook/lockbook/releases";
+    maintainers = [ maintainers.parth ];
```
[the remark this answers is not in the document]
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: ✅ 5a073b503d813580ee4ddf26beff5766db2656e2

**Review by pluiedev — APPROVED** [review]

**Comment by Parth** [comment]
Thank you for the thoughtful feedback @donovanglover & @pluiedev! Could you guys have another look? I believe I addressed all your feedback (and learned a lot along the way).

File: pkgs/by-name/lo/lockbook/package.nix [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  cargoHash = "sha256-+M+wL26KDbLKhcujPyWAsTlXwLrQVCUbTnnu/7sXul4=";
+
+  doCheck = false;
+  cargoBuildFlags = [
+    "--package"
+    "lockbook-cli"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
```
Comment (Aleksanaa) [inline pkgs/by-name/lo/lockbook/package.nix:28]:
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

File: pkgs/by-name/lo/lockbook/package.nix [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "lockbook";
+    repo = "lockbook";
+    rev = "refs/tags/${version}";
+    hash = "sha256-hqBjA/6MWlhVjV4m+cIcnoRTApHuzbPzivMsaQHfRcc=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-+M+wL26KDbLKhcujPyWAsTlXwLrQVCUbTnnu/7sXul4=";
+
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/by-name/lo/lockbook/package.nix:21]: Reason
```suggestion
  doCheck = false; # No checks defined
```

```suggestion
  doCheck = false; # Require network access
```

```suggestion
  doCheck = false; # Require instance of <specific service>
```

File: pkgs/by-name/lo/lockbook/package.nix [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  postInstall = ''
+    installShellCompletion --bash --name lockbook.bash <($out/bin/lockbook completions bash)
+    installShellCompletion --zsh --name _lockbook <($out/bin/lockbook completions zsh)
+    installShellCompletion --fish --name lockbook.fish <($out/bin/lockbook completions fish)
+  '';
+
+  meta = {
+    description = "Private, polished note-taking platform";
+    longDescription = ''
+      Write notes, sketch ideas, and store files in one secure place. Share seamlessly, keep data synced, and access it on any platform—even offline. Lockbook encrypts files so even we can’t see them, but don’t take our word for it: Lockbook is 100% open-source.
```
Comment (Aleksanaa) [inline pkgs/by-name/lo/lockbook/package.nix:37]:
```suggestion
      Write notes, sketch ideas, and store files in one secure place.
      Share seamlessly, keep data synced, and access it on any
      platform—even offline. Lockbook encrypts files so even we
      can’t see them, but don’t take our word for it: Lockbook is
      100% open-source.
```

File: pkgs/by-name/lo/lockbook/package.nix (hunk 10 above)
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: Thank you, I fixed & force pushed

File: pkgs/by-name/lo/lockbook/package.nix (hunk 11 above)
Comment (Parth) [reply pkgs/by-name/lo/lockbook/package.nix]: Thank you, I fixed & force pushed

**Comment by Parth** [comment]
Thanks @Aleksanaa, @donovanglover I believe I addressed all your comments, could you take another pass when you have some time?

Having some trouble understanding the permission hierarchy is there someone else I should be tagging for the final merge?

**Comment by Aleksanaa** [comment]
@ofborg build lockbook

**Pull request merged**